python/paddle/fluid/data_feeder.py

```diff
@@ -71,10 +71,20 @@ def _feed_impl_(self, data, lod, lod_level):
             for each_data in data:
                 self._feed_impl_(each_data, lod[1:], lod_level - 1)
 
+    def _check_shape_(self, shape):
+        for s1, s2 in zip(self.shape, shape):
+            if s1 != s2 and s1 >= 0 and s2 >= 0:
+                raise ValueError(
+                    "Shape not match. What is defined in data layer is {}, but receive {}".
+                    format(self.shape, shape))
+
     def done(self):
         arr = numpy.array(self.data, dtype=self.dtype)
-        if self.shape and len(arr.shape) != len(self.shape):
-            arr = arr.reshape(self.shape)
+        if self.shape:
+            if len(arr.shape) != len(self.shape):
+                arr = arr.reshape(self.shape)
```

Comment:
reshape can throw exception. Can catch this and throw a more user-friendly error message.

Reply:
Done.